plugin/storage/es/spanstore/reader.go

```diff
@@ -363,15 +364,18 @@ func (s *SpanReader) multiRead(ctx context.Context, traceIDs []model.TraceID, st
 		searchRequests := make([]*elastic.SearchRequest, len(traceIDs))
 		for i, traceID := range traceIDs {
 			query := buildTraceByIDQuery(traceID)
+			boolQuery := elastic.NewBoolQuery().Must(startTimeRangeQuery)
+			query = boolQuery.Must(query) // Adding time range query on startTimeMillis field.
 			if val, ok := searchAfterTime[traceID]; ok {
 				nextTime = val
 			}
 
-			s := s.sourceFn(query, nextTime)
-
-			searchRequests[i] = elastic.NewSearchRequest().
+			src := s.sourceFn(query, nextTime)
+			searchRequest := elastic.NewSearchRequest().
 				IgnoreUnavailable(true).
-				Source(s)
+				Source(src)
+
+			searchRequests[i] = searchRequest
```

Comment:
searchRequest: seems like unnecessary var. What was wrong with the original code?

Reply:
yes, internally we used it for logging so added the same here. Will remove this.